Misc fixes for running nmstatectl edit on Fedora.

cathay4t opened the pull request:
* Fix NULL interface name of NMSettingConnection.
 * Fix `unknown` duplex mode of virt-net.

tyll commented on libnmstate/nm/applier.py:
```diff
@@ -70,6 +70,9 @@ def prepare_edited_ifaces_configuration(ifaces_desired_state):
         if nmdev:
             cur_con_profile = connection.get_device_connection(nmdev)
+            if not cur_con_profile.get_interface_name():
+                set_conn = cur_con_profile.get_setting_connection()
```
Will `get_setting_connection()' always return a connection setting? Otherwise it might also need to be created. Another issue is that this might change unspecific connection profiles (not bound to an interface) that happen to be active for an interface to be specific for this interface. I am not sure if this is what nmstate should do or if nmstate should then copy the original profile and store it with the new settings and the interface name.

EdDev replied:
- Are you sure this is happening when the connection is in "activated" state? See #120 
- Why not update the new_con_profile with the interface if it is missing it? It sound safer to modify a new one over one which has been "read".

tyll commented on libnmstate/nm/wired.py:
```diff
@@ -66,7 +66,8 @@ def create_setting(iface_state, base_con_profile):
         wired_setting.props.speed = speed
 
-    if duplex:
+    if duplex != 'unknown':
```
This is not correct. This might be better:
```suggestion
    if duplex and duplex != 'unknown':
```

However, I guess the problem is what to do if auto-negotiation is reported as `false` but the device does not report a speed and duplex setting. Also I guess line 56 should be changed to not set `duplex` to `'unknown'` instead of checking it here.

cathay4t commented:
V2 patch uploaded.

Changes since V1:
 * Hide `ethernet` subtree if got strange duplex.
 * Add unit test for invalid duplex on `get_info()`.
 * Only set the interface name on duplicated new profile.
